packages/create-next-app/templates/app-tw/js/eslintrc.json

```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": [
+    "plugin:@typescript-eslint/recommended-type-checked",
```

Comment:
In the linked discussion #37151, I suggested creating a `next/typescript` preset. The advantage of that is:

* Having a single place that can store & customize the typescript-eslint recommendations
* Not farming out to an external tool (I do feel awkward putting my own project here...)

This PR doesn't (yet?) do that because:

* As of typescript-eslint v6, it's just one line to enable recommended rules with _and_ without type checking
* Creating a new `next/*` preset is a bit more plumbing 

Do you all have a preference?

Reply:
@JoshuaKGoldberg I spoke to @timneutkens and decided that we don't want `plugin:@typescript-eslint/recommended-type-checked` listed here, its should be a core feature of Next.js 

So if we can do it in the existing `next/core-web-vitals`, that would be ideal. However that might be a breaking change, so we could put it under `next/typescript` since these are TS only rules, right?

Reply:
Makes sense - will do!

Reply:
Done. I also updated `next lint` to add `'next/typescript'` to the `extends` values if a `'tsconfig.json'` file is found. Not sure that's the right way to check for whether they've enabled TypeScript?